libensemble/history.py

```diff
@@ -54,17 +54,25 @@ def __init__(self, alloc_specs, sim_specs, gen_specs, exit_criteria, H0):
         H = np.zeros(L + len(H0), dtype=dtype_list)
 
         if len(H0):
+            # Prepend H with H0
             fields = H0.dtype.names
 
             for field in fields:
                 H[field][:len(H0)] = H0[field]
                 # for ind, val in np.ndenumerate(H0[field]):  # Works if H0[field] has arbitrary dimension but is slow
                 #     H[field][ind] = val
 
-        # Prepend H with H0
-        H['sim_id'][:len(H0)] = np.arange(0, len(H0))
-        H['given'][:len(H0)] = 1
-        H['returned'][:len(H0)] = 1
+            if 'given' not in fields:
+                print("Marking entries in H0 as having been given and returned", flush=True)
+                H['given'][:len(H0)] = 1
+                H['returned'][:len(H0)] = 1
+            elif 'returned' not in fields:
+                print("Marking entries in H0 as having been returned", flush=True)
```

Comment:
I would make the print statements logger.manager_warning - thoes get echoed to std err by default. But we are bound by xsdk to not have print in core libe.

Reply:
I agree and have implemented this.